src/main/java/com/sublinks/sublinksapi/comment/services/CommentHistoryService.java

```diff
@@ -60,6 +60,9 @@ public void createOrAddCommentHistory(Comment comment) {
 
   @Transactional
   public int deleteAllByComment(Comment comment) {
+    if (!commentHistoryConfig.isKeepCommentHistory()) {
+      return 0;
+    }
```

Comment:
This should ignore the option, for the case where it got activated => comment was created / edited => deactivated